Bug: modulo operator accepts float operands

## Description
The modulo operator `%` accepts float operands at compile time but fails at runtime.

## Reproduction
```ez
import @std
using std

do main() {
    temp f float = 3.14
    temp x = f % 2
    println(x)
}
```

Passes `ez check` but fails at runtime with:
```
error[E3014]: unknown operator: FLOAT % INTEGER
```

## Expected Behavior
The type checker should validate that `%` is only used with integer operands.

--- a/integration-tests/fail/errors/E3002_modulo_float.ez
+++ b/integration-tests/fail/errors/E3002_modulo_float.ez
@@ -1,0 +1,12 @@
+/*
+ * Error Test: E3002 - modulo-float
+ * Expected: "%" and "integer"
+ */
+
+import @std
+using std
+
+do main() {
+    temp f float = 3.14
+    temp x = f % 2  // Should fail: modulo on float
+}

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -1805,7 +1805,7 @@ func (tc *TypeChecker) checkInfixExpression(infix *ast.InfixExpression) {
 			column,
 		)
 
-	case "-", "*", "/", "%":
+	case "-", "*", "/":
 		// Only valid for numbers
 		if !tc.isNumericType(leftType) || !tc.isNumericType(rightType) {
 			tc.addError(
@@ -1834,6 +1834,17 @@ func (tc *TypeChecker) checkInfixExpression(infix *ast.InfixExpression) {
 					column,
 				)
 			}
+		}
+
+	case "%":
+		// Modulo only valid for integers (#601)
+		if !tc.isIntegerType(leftType) || !tc.isIntegerType(rightType) {
+			tc.addError(
+				errors.E3002,
+				fmt.Sprintf("invalid operands for '%%': %s and %s (expected integer)", leftType, rightType),
+				line,
+				column,
+			)
 		}
 
 	case "==", "!=":

